### PR TITLE
Introduce max_id_ever_seen state

### PR DESCRIPTION
**Pull request by pwaller**
we now compute, store, and later use, a `max_id_ever_seen` state.

### CODE REVIEW

**Comment by pwaller** [comment]
So now we think we're done. It's ugly because we don't have a proper variable saving mechanism.

**Comment by pwaller** [comment]
/cc @frabcus 

We're not 100% confident it fixes the issue because we've only done cursory performance testing. But it appears to have the right effect.

**Comment by frabcus** [comment]
:thumbsup:

File: twsearch.py [hunk 1]
```diff
@@ -196,6 +196,35 @@ def change_window(start, end):
...
+    """
+    try:
+	return scraperwiki.sql.select('max_id_seen from __max_id')[0]['max_id_seen']
+    except sqlite3.OperationalError:
+        return get_max_id_ever_seen_expensive()
+ 
+
+def set_max_id_ever_seen(max_id_ever_seen):
+    data = {'id': 'max_id_seen', 'max_id_seen': max_id_ever_seen}
+    scraperwiki.sql.save(['id'], data, table_name='__max_id')
```
Comment (drj11) [inline twsearch.py:225]: we need to remove this table in the `command_clean_state()` function, which we don't do. yet.

If we had decent persistent variables, i wouldn't be writing this comment.

**Comment by pwaller** [comment]
@frabcus we've deployed this. Grove appears to have suffered a netsplit so we can't see you on IRC at the moment.

**Comment by pwaller** [comment]
As @drj11 notes, we won't get any benefit until at least one process manages to complete the slow route. And they're all still contending reads, so this might take a while.

**Pull request merged**